### DISCUSSION TITLE
Interactive rollback

### DISCUSSION THREAD

**Discussion by bilogic** [deployphp/deployer#2569] [Help needed]
Hi,

Sorry if there are several questions, but I think answering all of them together will provide a holistic solution.

Let's say after deployment, `current` -> `releases/3/`, and I have a `releases/1/` and `releases/2/`

1. How do I perform a rollback?
2. Is there a way to make it interactive? i.e. List out what's available and let user pick?
3. After rolling back, is there a way to "delete" the problematic deployment, `3`?
4. Is there a way to rollback to `1` and delete `2` and `3`?
5. If `2` and `3` are deleted, will it still be `2` the next deployment? (Just for my info)

Example code will be greatly appreciated! Thank you.

**Comment by antonmedv** [comment]
I’m writing from a phone. Sorry for formatting. 

1. dep rollback
2. There is no interactive mode (actually nice idea). But you can list available releases: dep status. 
3. Deployer will mark problematic release and will avoid it in future. Delete via: dep run rm -rf releases/3
4. Run dep rollback twice or symlink correct release via dep run. 
5. If I remember correctly no, it will be 4.

**Reply by bilogic** [reply]
Thanks for the prompt answers!
